src/routes/usuario.js

```diff
@@ -1,96 +1,17 @@
 const express = require('express');
-const bcrypt = require('bcrypt');
-const pool = require('../utils/database');
-const logger = require('../utils/logger');
+const LoginValidation = require('../validations/loginValidation');
+const SignupValidation = require('../validations/signupValidation');
+const userController = require('../controllers/userController');
+
+const loginValidation = new LoginValidation();
+const signupValidation = new SignupValidation();
 
-const saltRounds = 10;
 const router = express.Router();
 
-router.post('/login', (req, res) => {
-  (async () => {
-    const client = await pool.connect();
-    try {
-      const resq = await client.query('SELECT password_hash FROM users WHERE username = $1', [req.body.username]);
-      const match = await bcrypt.compare(req.body.password, resq.rows[0].password_hash);
-      if (match) {
-        req.session.user = {
-          username: req.body.username,
-          email: req.body.email,
-          password: resq.rows[0].password_hash,
-        };
-        res.sendStatus(200);
-      } else {
-        res.sendStatus(400);
-      }
-    } finally {
-      client.release();
-    }
-  })().catch((e) => {
-    logger.info(e);
-    res.sendStatus(500);
-  });
-});
+router.post('/login', loginValidation.validateInput, loginValidation.validateUsernamePassword, userController.login);
```

Comment:
Validations should be done in controller (calling its corresponding Validator). Avoid using middlewares like this, they tend to be a pain in the ass for debugging.